src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -265,14 +369,46 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             CSharpCompilation compilation,
             MethodSymbol method,
             BoundNode node,
-            DiagnosticBag diagnostics,
-            Action<BoundExpression, TypeSymbolWithAnnotations> callbackOpt = null)
+            DiagnosticBag diagnostics)
         {
             if (method.IsImplicitlyDeclared && (!method.IsImplicitConstructor || method.ContainingType.IsImplicitlyDeclared))
             {
                 return;
             }
-            Analyze(compilation, method, node, diagnostics, useMethodSignatureReturnType: false, useMethodSignatureParameterTypes: false, methodSignatureOpt: null, returnTypes: null, initialState: null, callbackOpt);
+            Analyze(compilation,
+                method,
+                node,
+                diagnostics,
+                useMethodSignatureReturnType: false,
+                useMethodSignatureParameterTypes: false,
+                methodSignatureOpt: null,
+                returnTypes: null,
+                initialState: null);
+        }
+
+        internal static BoundNode AnalyzeAndRewrite(
+            CSharpCompilation compilation,
+            MethodSymbol method,
+            BoundNode node,
+            DiagnosticBag diagnostics)
+        {
+            var analyzedNullabilities = PooledDictionary<BoundExpression, (NullabilityInfo, TypeSymbol)>.GetInstance();
+            Analyze(
+                compilation,
+                method,
+                node,
+                diagnostics,
+                useMethodSignatureReturnType: false,
+                useMethodSignatureParameterTypes: false,
+                methodSignatureOpt: null,
+                returnTypes: null,
+                initialState: null,
+                analyzedNullabilityMapOpt: analyzedNullabilities);
+#if DEBUG
+            // If the map was provided, we're in a nested call so we can't yet verify that all things are in the map.
```

Comment:
>// If the map was provided, we're in a nested call so we can't yet verify that all things are in the map. [](start = 12, length = 105)

The map can no longer be passed into `AnalyzeAndRewrite`. Comment can be removed?